java/java.hints/src/org/netbeans/modules/java/hints/errors/Utilities.java

```diff
@@ -3265,16 +3274,36 @@ public Void visitIdentifier(IdentifierTree node, Void p) {
                 }
             }
         }
-        if (isReturnExpression) {
-            ExpressionTree et = (ExpressionTree) make.SwitchExpression(TreeShims.getExpressions(st).get(0), newCases);
-            wc.rewrite(st, make.Return(et));
-        } else if (isExpression) {
-            ExpressionTree et = (ExpressionTree) make.SwitchExpression(TreeShims.getExpressions(st).get(0), newCases);
-            wc.rewrite(st, make.ExpressionStatement((ExpressionTree) make.Assignment((ExpressionTree) variable, et)));
-        } else if (switchExpressionFlag) {
-            wc.rewrite(st, make.SwitchExpression(TreeShims.getExpressions(st).get(0), newCases));
-        } else {
-            wc.rewrite((SwitchTree) st, make.Switch(((SwitchTree) st).getExpression(), newCases));
+        ExpressionTree et = null;
+        switch(switchType){
+            case SWITCH_EXPRESSION_RETURN_CASE:
+                et = (ExpressionTree) make.SwitchExpression(TreeShims.getExpressions(st).get(0), newCases);
+                if(typeCastTree != null){
+                    et = make.Parenthesized(et);
+                    et = make.TypeCast(typeCastTree, et);
+                }
+                wc.rewrite(st, make.Return(et));
+                break;
+            case SWITCH_EXPRESSION_CAST_CASE:
+                et = (ExpressionTree) make.SwitchExpression(TreeShims.getExpressions(st).get(0), newCases);
+                if(typeCastTree != null){
+                    et = make.Parenthesized(et);
+                    et = make.TypeCast(typeCastTree, et);
+                }
+                wc.rewrite(st, make.ExpressionStatement((ExpressionTree) make.Assignment((ExpressionTree) leftVariable, et)));
+                break;
+            case SWITCH_EXPRESSION_ASSIGN_CASE:
+                et = (ExpressionTree) make.SwitchExpression(TreeShims.getExpressions(st).get(0), newCases);
+                wc.rewrite(st, make.ExpressionStatement((ExpressionTree) make.Assignment((ExpressionTree) leftVariable, et)));
+                break;
+            case SWITCH_EXPRESSION:
```

Comment:
IMHO cases for `SWITCH_EXPRESSION_RETURN_CASE`, `SWITCH_EXPRESSION_CAST_CASE`, `SWITCH_EXPRESSION_ASSIGN_CASE` can be mostly merged, except the last rewrite step, which is unique for each of them.
- all are switch expressions;
- if `typeCastTree` is available, wrap in type cast
- (the case-specific statement replacement)